@xen-orchestra/web-core/lib/components/button/UiButton.vue

```diff
@@ -0,0 +1,207 @@
+    color: 'info',
+    level: 'primary',
+    size: 'small',
+  }
+)
+
+const isDisabled = useContext(DisabledContext, () => props.disabled)
+
+const fontClasses = {
+  'extra-small': 'typo p3-medium',
+  small: 'typo h6-medium',
+  medium: 'typo h3-semi-bold',
+}
+
+const fontClass = computed(() => fontClasses[props.size])
+
+const className = computed(() => {
+  return [
+    props.color,
+    props.level,
+    props.size,
+    fontClass.value,
+    {
+      busy: props.busy,
+      disabled: isDisabled.value,
+    },
+  ]
+})
+</script>
+
+<style lang="postcss" scoped>
+/* COLOR VARIANTS */
+.ui-button {
+  &.info {
+    --accent-color: var(--color-purple-base);
+
+    &:is(:hover, .hover, :focus-visible) {
+      --accent-color: var(--color-purple-d20);
+    }
+
+    &:is(:active, .pressed) {
+      --accent-color: var(--color-purple-d40);
+    }
+
+    &:is(:disabled, .disabled) {
+      --accent-color: var(--color-grey-400);
+    }
+  }
+
+  &.success {
+    --accent-color: var(--color-green-base);
+
+    &:is(:hover, .hover, :focus-visible) {
+      --accent-color: var(--color-green-d20);
+    }
+
+    &:is(:active, .pressed) {
+      --accent-color: var(--color-green-d40);
+    }
+
+    &:is(:disabled, .disabled) {
+      --accent-color: var(--color-green-l60);
+    }
+  }
+
+  &.warning {
+    --accent-color: var(--color-orange-base);
+
+    &:is(:hover, .hover, :focus-visible) {
+      --accent-color: var(--color-orange-d20);
+    }
+    &:is(:active, .pressed) {
+      --accent-color: var(--color-orange-d40);
+    }
+
+    &:is(:disabled, .disabled) {
+      --accent-color: var(--color-orange-l60);
+    }
+  }
+
+  &:is(.danger, .error) {
+    --accent-color: var(--color-red-base);
+
+    &:is(:hover, .hover, :focus-visible) {
+      --accent-color: var(--color-red-d20);
+    }
+
+    &:is(:active, .pressed) {
+      --accent-color: var(--color-red-d40);
+    }
+    &:is(:disabled, .disabled) {
+      --accent-color: var(--color-red-l60);
+    }
+  }
+}
+
+/* SIZE VARIANTS */
+.ui-button {
+  &.extra-small {
+    --padding: 0.4rem 0.8rem;
+
+    &.tertiary {
+      --padding: 0;
+    }
+  }
+
+  &.small {
+    --padding: 0.8rem 1.6rem;
+
+    &.tertiary {
+      --padding: 0.2rem 0;
+    }
+  }
+
+  &.medium {
+    --padding: 1.6rem 2.4rem;
+
+    &.tertiary {
+      --padding: 0.4rem 0;
+    }
+  }
+}
+
+/* LEVELS VARIANTS */
+.ui-button {
+  & {
+    --border-radius: 0.8rem;
+    --border-style: solid;
+  }
+
+  &.primary {
+    --color: var(--color-grey-600);
+    --background-color: var(--accent-color);
+    --border-color: var(--accent-color);
+  }
+
+  &.secondary {
+    --color: var(--accent-color);
+    --background-color: var(--background-color-primary);
+    --border-color: var(--accent-color);
+  }
+
+  &.tertiary {
+    --color: var(--accent-color);
+    --background-color: transparent;
+    --border-color: var(--accent-color);
+    --border-style: none none solid;
+    --border-radius: 0;
+  }
+}
```

Comment:
As discussed in the meantime, it'll be more legible to specify every variations in each variant.

```suggestion
.ui-button {
  &.primary {
    --color: var(--color-grey-600);
    --background-color: var(--accent-color);
    --border-color: var(--accent-color);
    --border-style: solid;
    --border-radius: 0.8rem;
  }

  &.secondary {
    --color: var(--accent-color);
    --background-color: var(--background-color-primary);
    --border-color: var(--accent-color);
    --border-style: solid;
    --border-radius: 0.8rem;
  }

  &.tertiary {
    --color: var(--accent-color);
    --background-color: transparent;
    --border-color: var(--accent-color);
    --border-style: none none solid;
    --border-radius: 0;
  }
}
```